### ISSUE TITLE
UPNP_DISCOVERY_SEARCH_TIMEOUT; IPv6 compatibility issue?

### ISSUE THREAD

**Issue by eddyg** [philippe44/AirConnect#66]
I am unable to get my Linux server (running Ubuntu 14.04.5 LTS) to discover any devices:

```
server:~> ./airupnp-x86-64 -i airupnp.xml -d all=sdebug
[17:44:38.290743] main:1205 Starting airupnp version: v0.2.0.5 (Mar 12 2018 @ 07:25:18)
[17:44:38.290893] main:1213 no config file, using defaults
[17:44:38.291361] Start:965 Binding to 10.1.1.1:49152
[17:44:58.006498] MasterHandler:504 event: 7 [UPNP_DISCOVERY_SEARCH_TIMEOUT] [(nil)] (recurse 0)
[17:44:58.006549] UpdateThread:643 Presence checking
[17:44:59.491852] Stop:1006 terminate update thread ...
[17:44:59.492008] Stop:1011 terminate main thread ...
[17:44:59.492051] Stop:1016 flush renderers ...
[17:44:59.492067] Stop:1019 terminate libupnp
```

Could this be related to having IPv6 active and enabled on the NICs? I ask because [this comment](https://bugs.launchpad.net/ubuntu/+source/libupnp/+bug/1571199/comments/14) from the `libupnp` project talks about disabling IPv6 as a "hack" to get it to work, and am curious if this could also affect `airupnp`.

I'm unable to disable IPv6 on this server to confirm that this is definitely the issue, but I tried `airupnp` on a different Ubuntu server (that is also running 14.04.5 LTS) which does *not* have any IPv6 stuff configured and it started up straight away.

**Comment by philippe44** [comment]
it should not - I have IPv6 on my system and it works. Do you have a firewall? Are you sure airupnp is binding to the right interface?

**Comment by eddyg** [comment]
The server it was failing on does have two NICs in a common two-interface firewall config with IP masquerading, along with firewall rules configured via `iptables` and `ip6tables`. One of the first things I did was check the firewall logs, but nothing was recorded during `airupnp`'s discovery failures.

The debug output does show it binding to the "private side" NIC address like it should be.

Since this server is running an old version of Ubuntu, it's probably not worth spending a lot of time on at this point. 18.04 will be out soon, and I plan to upgrade in a couple months and can re-visit. The IPv6 thing was just a wild guess based on some quick googling.

In the mean time, feel free to close this issue, and accept my thanks for your prompt reply and for a _very_ useful tool!

**Comment by philippe44** [comment]
You have the -b option to bind airupnp to a specific interface

**Comment by eddyg** [comment]
Right... and during my testing I did try explicitly specifying the address of the private-side NIC via `-b`, but the behavior was the same.

**Comment by philippe44** [comment]
oh, I mis-read your message. If you do masquerading, it can't work. Answers won't be received

**Comment by eddyg** [comment]
To clarify, `eth0` is the "private side" of the two-interface firewall, and `eth1` is "public side". The private side only masquerades to public side when it has to communicate to "the outside world". There shouldn't be any reason that `airupnp`, bound to an address on `eth0` (10.1.1.1  on the private side), can't "discover" devices that are also in that subnet on the private side. The discovery is not crossing the masquerading boundary.

Obviously there is something weird going on with my system since this works for you, and I will dig deeper in to this if the problem persists after upgrading to 18.04 in a few months.

**Comment by philippe44** [comment]
So just to be clear, all other UPnP devices are on the same subnet and there is no natting happening between airupnp and UPnP devices, right? Why not trying to run an upnp renderer on the same machine first to see if it is discovered when is on localhost at least?

**Comment by meiser79** [comment]
I can reproduce the issue on my laptop and Synology NAS, both running in dual-stack mode. Even by binding it to an IP address, it's not working. Any help is appreciated.

```
$ ./airupnp-x86-64 -i test.xml
[17:57:17.817591] main:1205 Starting airupnp version: v0.2.0.5 (Mar 12 2018 @ 07:25:18)
[17:57:17.817681] main:1213 no config file, using defaults
[17:57:17.819542] Start:965 Binding to 192.168.3.8:49152
[17:57:39.021725] Stop:1006 terminate update thread ...
[17:57:39.021841] Stop:1011 terminate main thread ...
[17:57:39.021867] Stop:1016 flush renderers ...
[17:57:39.021884] Stop:1019 terminate libupnp
$ ./airupnp-x86 -i test.xml
[17:57:51.102040] main:1205 Starting airupnp version: v0.2.0.5 (Mar 12 2018 @ 07:25:07)
[17:57:51.102137] main:1213 no config file, using defaults
[17:57:51.105217] Start:965 Binding to 192.168.3.8:49152
[17:58:12.306033] Stop:1006 terminate update thread ...
[17:58:12.306182] Stop:1011 terminate main thread ...
[17:58:12.306213] Stop:1016 flush renderers ...
[17:58:12.306227] Stop:1019 terminate libupnp
```
